### ISSUE TITLE
Implement `module.id`

### ISSUE THREAD

**Issue by hkoba** [oven-sh/bun#2195]
### What is the problem this feature would solve?

In Node, we can write a module that also can be used as a runnable script using ["`module.id`"](https://nodejs.org/dist/latest-v18.x/docs/api/modules.html#moduleid) like the following:

```javascript
export function foo() { ... }

if (module.id === '.') {
   console.log(foo())
}
```

Some users may prefer the use of `module.filename === __filename`.

We must write another script file only to import and run the module when we have neither of them.

### What is the feature you are proposing to solve the problem?

This will help us write a module that also can be used as a runnable script.

Such a feature is widely used in other languages too. See [this SO entry](https://stackoverflow.com/questions/419163/what-does-if-name-main-do) and GitHub search results of [Python](https://github.com/search?q=if+__name__+%3D%3D+%22__main__%22+extension%3Apy+language%3APython&type=Code&ref=advsearch&l=Python&l=) and [Ruby](https://github.com/search?q=%22if+__FILE__+%3D%3D%22+extension%3Arb+language%3ARuby&type=Code&ref=advsearch&l=Ruby&l=).

### What alternatives have you considered?

I don't know whether Bun has already provided a way to examine such an invocation context. But I expect exposing "module.id" matches better for Bun's goal than inventing another way.

**Comment by Jarred-Sumner** [comment]
The bun-specific way of doing this is

`import.meta.path` or `import.meta.url`

We need to support this for Node compatibility though

**Comment by hkoba** [comment]
Sorry, I was wrong about the example of `module.filename` in the description of this feature request.

> 
> Some users may prefer the use of `module.filename === __filename`.
> 

The above expression is probably always true. What I should write was the following:

```javascript
if (module.filename === process.argv[1]) {
  console.log(`${__filename} is the main module`)
}
```

**Comment by simylein** [comment]
I think this is supported now as running:

```ts
if (module.filename === process.argv[1]) {
	console.log(`${__filename} is the main module`);
}
```

yields
```
/Users/simylein/main.ts is the main module
```

I am using bun version 0.7.1

If it works for you, can we close this as resolved?

**Comment by Jarred-Sumner** [comment]
`module.id` was implemented at runtime in Bun v0.6.x
